docs/book/how-to/develop-locally/local-prod-pipeline-variants.md

```diff
@@ -0,0 +1,135 @@
+---
+description: Create different variants of your pipeline for local development and production.
+---
+
+# Create pipeline variants for local development and production
+
+When developing ZenML pipelines, it's often beneficial to have different variants of your pipeline for local development and production environments. This approach allows you to iterate quickly during development while maintaining a full-scale setup for production. While configuration files are one way to achieve this, you can also implement this directly in your code.
+
+## Creating pipeline variants
+
+There are several ways to create different variants of your pipeline:
+
+1. Using configuration files
+2. Implementing variants in code
+3. Using environment variables
+
+Let's explore each of these methods:
+
+### Using configuration files
+
+ZenML allows you to specify pipeline and step configurations using YAML files. Here's an example:
+
+```yaml
+enable_cache: False
+parameters:
+    dataset_name: "small_dataset"
+steps:
+    load_data:
+        enable_cache: False
+```
+
+This config file sets up a development variant by using a smaller dataset and disabling caching.
+
+To apply this configuration to your pipeline, use the `with_options(config_path=<PATH_TO_CONFIG>)` pattern:
+
+```python
+from zenml import step, pipeline
+
+@step
+def load_data(dataset_name: str) -> dict:
+    ...
+
+@pipeline
+def ml_pipeline(dataset_name: str):
+    load_data(dataset_name)
+
+if __name__ == "__main__":
+    ml_pipeline.with_options(config_path="path/to/config.yaml")()
+```
+
+You can create separate configuration files for development and production:
+
+- `config_dev.yaml`: Configuration for local development
+- `config_prod.yaml`: Configuration for production
+
+### Implementing variants in code
+
+You can also create pipeline variants directly in your code:
+
+```python
+import os
+from zenml import step, pipeline
+
+@step
+def load_data(dataset_name: str) -> dict:
+    # Load data based on the dataset name
+    ...
+
+@pipeline
+def ml_pipeline(is_dev: bool = False):
+    dataset = "small_dataset" if is_dev else "full_dataset"
+    load_data(dataset)
+
+if __name__ == "__main__":
+    is_dev = os.environ.get("ENVIRONMENT") == "dev"
```

Comment:
maybe use ZENML_ENVIRONMENT as the ENVIRONMENT env variable might be overloaded by other tools